app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,68 @@
+class MarketsController < ApplicationController
+  def index
+    @markets = Market.all
+  end
+
+  def login
+    id = params.permit(:login_id)[:login_id].to_i
+
+    if (id <= Market.last.id && id > 0)
+      redirect_to "/markets/#{ id }/dashboard"
+    else
+      redirect_to "/markets/market_not_found"
+    end
+  end
+
+  def dashboard
+    @market = Market.find(params[:id])
```

Comment:
What happens if someone types `/markets/8000000000/dashboard` into their browser's URL bar? You have guards against someone logging in with a non-existent market, but not against them typing it into their browser directly.